common/djangoapps/track/tests/test_segment.py

```diff
@@ -0,0 +1,150 @@
+        ({'page': sentinel.page}, {'page': {'url': sentinel.provided_page}}, {'page': {'url': sentinel.page}}),
+        ({'client_id': sentinel.client_id}, {'Google Analytics': {'clientId': sentinel.provided_client_id}}, {'Google Analytics': {'clientId': sentinel.client_id}}),
+    )
+    @override_settings(LMS_SEGMENT_KEY="testkey")
+    def test_track_context_with_stuff(self, tracking_context, provided_context, expected_segment_context):
+        # Test first with tracking and no provided context.
+        with self.tracker.context('test', tracking_context):
+            segment.track(sentinel.user_id, sentinel.name, self.properties)
+        args, kwargs = self.mock_segment_track.call_args
+        self.assertEqual((sentinel.user_id, sentinel.name, self.properties, expected_segment_context), args)
+
+        # Test with provided context and no tracking context.
+        segment.track(sentinel.user_id, sentinel.name, self.properties, provided_context)
+        args, kwargs = self.mock_segment_track.call_args
+        self.assertEqual((sentinel.user_id, sentinel.name, self.properties, provided_context), args)
+
+        # Test with provided context and also tracking context.
+        with self.tracker.context('test', tracking_context):
+            segment.track(sentinel.user_id, sentinel.name, self.properties, provided_context)
+        self.assertTrue(self.mock_segment_track.called)
+        args, kwargs = self.mock_segment_track.call_args
+        self.assertEqual((sentinel.user_id, sentinel.name, self.properties, provided_context), args)
+
+    @override_settings(LMS_SEGMENT_KEY="testkey")
+    def test_track_with_standard_context(self):
+
+        # Note that 'host' and 'path' will be urlparsed, so must be strings.
+        tracking_context = {
+            'accept_language': sentinel.accept_language,
+            'referer': sentinel.referer,
+            'username': sentinel.username,
+            'session': sentinel.session,
+            'ip': sentinel.ip,
+            'host': 'hostname',
+            'agent': sentinel.agent,
+            'path': '/this/is/a/path',
+            'user_id': sentinel.user_id,
+            'course_id': sentinel.course_id,
+            'org_id': sentinel.org_id,
+            'client_id': sentinel.client_id,
+        }
+        with self.tracker.context('test', tracking_context):
+            segment.track(sentinel.user_id, sentinel.name, self.properties)
+
+        self.assertTrue(self.mock_segment_track.called)
+        args, kwargs = self.mock_segment_track.call_args
+
+        expected_segment_context = {
+            'ip': sentinel.ip,
+            'Google Analytics': {
+                'clientId': sentinel.client_id,
+            },
+            'userAgent': sentinel.agent,
+            'page': {
+                'path': '/this/is/a/path',
+                'referrer': sentinel.referer,
+                'url': '//hostname/this/is/a/path'  # Synthesized URL value.
```

Comment:
It seems to work in GA, from my testing over the weekend.    It seemed safer to me to exclude the scheme when I didn't know it rather than arbitrarily choosing http or https.